eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -150,8 +158,15 @@ jobs:
   - script: |
       set -x
 
-      if [[ -z '${{ parameters.reuseBuildArtifactsFrom }}' ]]; then
-        docker run --rm -v "$(sourcesPath):/vmr" -w /vmr ${{ parameters.container }} ./prep.sh
+      if [[ -z '${{ parameters.reuseBuildArtifactsFrom }}' || '${{ parameters.prepNoArtifacts }}' == 'True' || '${{ parameters.prepNoSdk }}' == 'True' ]]; then
+        customPrepArgs=""
+        if [[ '${{ parameters.prepNoArtifacts }}' == 'True' ]]; then
+          customPrepArgs="${customPrepArgs} --no-artifacts"
+        fi
+        if [[ '${{ parameters.prepNoSdk }}' == 'True' ]]; then
+          customPrepArgs="${customPrepArgs} --no-sdk"
+        fi
+        docker run --rm -v "$(sourcesPath):/vmr" -w /vmr ${{ parameters.container }} ./prep.sh ${customPrepArgs}
       else
         mkdir $(sourcesPath)/.dotnet
```

Comment:
With these changes, it looks like we are skipping over the step to extract the previous SDK if noArtifacts or NoSdk is specified.

Reply:
It looks like CI just failed because of this - https://dev.azure.com/dnceng/internal/_build/results?buildId=2129748&view=logs&j=775596e9-8dfa-54d6-c033-e0a83b7464ac&t=05df5f54-d15a-5264-a7f0-71ac2ba0087e

Reply:
I think my change should fix this - extract the previous SDK if we have `reuseBuildArtifactsFrom` or `NoSdk`.  Running in https://dev.azure.com/dnceng/internal/_build/results?buildId=2129923&view=logs&j=61a52f3a-cec1-5bff-8c02-d8ba2c8bed99&t=95482b87-4101-575b-7ca3-baea90ec4b29 now.

Reply:
Agreed this should work but I still think when backporting the individual prepNoArtifacts and prepNoSdk should be inferred off of the existing parameters.